src/final_train_script.py

```diff
@@ -86,6 +86,30 @@ def training(model, train_dataloader, val_dataloader, num_epochs, lr,
         torch.save(best_model_para, os.path.join("weights", f'{datetime.strftime(datetime.now(), "%B-%d-%H")}.pt'))
         print(f'Best Validation Acc: {best_validation_acc}')
 
+def prediction(test_dataloader,model, weights):
+    nick_name = "tuge0"
+    weights_path = os.path.join("weights", weights)
+    if not os.path.isfile(weights_path):
+        raise Exception("weights file not found")
+    model.cuda()
+    model.eval()
+    model.load_state_dict(torch.load(weights_path))
+    result = []
+    with tqdm(test_dataloader, unit='img') as pbar:
+        for img in pbar:
+            with torch.no_grad():
+                pred_mask = model(img.to(DEVICE))
+                if len(result) == 0:
+                    # result = torch.unsqueeze(pred_mask, 0)
+                    result = torch.squeeze(pred_mask,1)
+                else:
+                    result = torch.cat((result, pred_mask), 0)
+    print(f'prediction result: {result.shape}')
+    sigmoid = torch.sigmoid(result)
+    sigmoid[result > 0.5] = 1
+    sigmoid[result <= 0.5] = 0
```

Comment:
The output of the network is not binary class. So no sigmoid here. The output is depth estimation. Try to find some example and take a look at theirs